### DISCUSSION TITLE
Rerendering before data responce

### DISCUSSION THREAD

**Discussion by dominicboston-kaluza** [TanStack/query#3573] [Q&A] [answered]
I'm struggling to find the right bit of docs to help me answer this question. 

When useQuery has received data is caches it and will return it without calling the endpoint again, great!
What happens when the data from the endpoint hasn't been received back yet and a rerender occurs. I kinda just assume it ignores the repeated request and doesn't make another call.

If anyone has the answer or can point me to a doc page that would be great.

Thanks
Dom

**Comment by TkDodo** [comment, marked as answer]
> What happens when the data from the endpoint hasn't been received back yet and a rerender occurs. I kinda just assume it ignores the repeated request and doesn't make another call.

Yes, that's right. If a request is in-flight, all `useQuery` instances (even those that mount afterwards) will pick up the running Promise.

**Reply by dominicboston-kaluza** [reply]
Thanks!
